Subject: [PATCH] svlogd: fix -Wextra sign-compare

- dirn loops: cast to int for the int counters (small dir count)
- n/j entry counts: cast to unsigned vs nmax/nmin (original
  promoted semantics, counts are non-negative)
- fwrite result vs unsigned len: cast len to int
diff --git a/runit/svlogd.c b/runit/svlogd.c
index f7576f0..b613316 100644
--- a/runit/svlogd.c
+++ b/runit/svlogd.c
@@ -533,7 +533,7 @@ static void rmoldest(struct logdir *ld)
 		warn2("can't read directory", ld->name);
 	closedir(d);
 
-	if (ld->nmax && (n > ld->nmax)) {
+	if (ld->nmax && ((unsigned)n > ld->nmax)) {
 		if (verbose)
 			bb_error_msg(INFO"delete: %s/%s", ld->name, oldest);
 		if ((*oldest == '@') && (unlink(oldest) == -1))
@@ -624,7 +624,7 @@ static int buffer_pwrite(int n, char *s, unsigned len)
 		////i = full_write(ld->fdcur, s, len);
 		////if (i != -1) break;
 		i = fwrite(s, 1, len, ld->filecur);
-		if (i == len) break;
+		if (i == (int)len) break;
 
 		if ((errno == ENOSPC) && (ld->nmin < ld->nmax)) {
 			DIR *d;
@@ -651,7 +651,7 @@ static int buffer_pwrite(int n, char *s, unsigned len)
 					ld->name);
 			closedir(d);
 			errno = ENOSPC;
-			if (j > ld->nmin) {
+			if ((unsigned)j > ld->nmin) {
 				if (*oldest == '@') {
 					bb_error_msg(WARNING"out of disk space, delete: %s/%s",
 							ld->name, oldest);
@@ -879,7 +879,7 @@ static void logdirs_reopen(void)
 	int ok = 0;
 
 	tmaxflag = 0;
-	for (l = 0; l < dirn; ++l) {
+	for (l = 0; l < (int)dirn; ++l) {
 		logdir_close(&dir[l]);
 		if (logdir_open(&dir[l], fndir[l]))
 			ok = 1;
@@ -911,7 +911,7 @@ static int buffer_pread(/*int fd, */char *s, unsigned len)
 
 	do {
 		if (rotateasap) {
-			for (i = 0; i < dirn; ++i)
+			for (i = 0; i < (int)dirn; ++i)
 				rotate(dir + i);
 			rotateasap = 0;
 		}
@@ -926,7 +926,7 @@ static int buffer_pread(/*int fd, */char *s, unsigned len)
 		}
 		now = monotonic_sec();
 		nearest_rotate = now + (45 * 60 + 45);
-		for (i = 0; i < dirn; ++i) {
+		for (i = 0; i < (int)dirn; ++i) {
 			if (dir[i].rotate_period) {
 				if (LESS(dir[i].next_rotate, now))
 					rotate(dir + i);
@@ -999,7 +999,7 @@ static void sig_child_handler(int sig_no UNUSED_PARAM)
 	if (verbose)
 		bb_error_msg(INFO"sig%s received", "child");
 	while ((pid = wait_any_nohang(&wstat)) > 0) {
-		for (l = 0; l < dirn; ++l) {
+		for (l = 0; l < (int)dirn; ++l) {
 			if (dir[l].ppid == pid) {
 				dir[l].ppid = 0;
 				processorstop(&dir[l]);
@@ -1092,7 +1092,7 @@ int svlogd_main(int argc, char **argv)
 	fdwdir = xopen(".", O_RDONLY|O_NDELAY);
 	close_on_exec_on(fdwdir);
 	dir = xzalloc(dirn * sizeof(dir[0]));
-	for (i = 0; i < dirn; ++i) {
+	for (i = 0; i < (int)dirn; ++i) {
 		dir[i].fddir = -1;
 		dir[i].fdcur = -1;
 		////dir[i].btmp = xmalloc(buflen);
@@ -1190,7 +1190,7 @@ int svlogd_main(int argc, char **argv)
 			memcpy(printptr, stamp, 25);
 			printptr[25] = ' ';
 		}
-		for (i = 0; i < dirn; ++i) {
+		for (i = 0; i < (int)dirn; ++i) {
 			struct logdir *ld = &dir[i];
 			if (ld->fddir == -1)
 				continue;
@@ -1224,7 +1224,7 @@ int svlogd_main(int argc, char **argv)
 				ch = lineptr[linelen-1];
 			}
 			/* linelen == no of chars incl. '\n' (or == stdin_cnt) */
-			for (i = 0; i < dirn; ++i) {
+			for (i = 0; i < (int)dirn; ++i) {
 				if (dir[i].fddir == -1)
 					continue;
 				if (dir[i].matcherr == 'e') {
@@ -1251,7 +1251,7 @@ int svlogd_main(int argc, char **argv)
 		fflush_all();////
 	}
 
-	for (i = 0; i < dirn; ++i) {
+	for (i = 0; i < (int)dirn; ++i) {
 		if (dir[i].ppid)
 			while (!processorstop(&dir[i]))
 				continue;
